kernel/src/process_binary.rs

```diff
@@ -114,15 +114,15 @@ impl fmt::Debug for ProcessBinaryError {
 pub struct ProcessBinary {
     /// Process flash segment. This is the entire region of nonvolatile flash
     /// that the process occupies.
-    pub(crate) flash: &'static [u8],
+    pub flash: &'static [u8],
 
     /// The footers of the process binary (may be zero-sized), which are metadata
     /// about the process not covered by integrity. Used, among other things, to
     /// store signatures.
-    pub(crate) footers: &'static [u8],
+    pub footers: &'static [u8],
 
     /// Collection of pointers to the TBF header in flash.
-    pub(crate) header: tock_tbf::types::TbfHeader,
+    pub header: tock_tbf::types::TbfHeader,
```

Comment:
While I don't think there's necessarily anything in a read-only reference to process flash that needs to be protected, this is a pretty big change in visibility. My instinct is that it's preferred to prevent any capsule from accessing this more willy-nilly.

Having only googled [this](https://users.rust-lang.org/t/make-struct-fields-visible-to-selected-modules/73757/3) briefly, can we use the [`pub (in path)`](https://doc.rust-lang.org/reference/visibility-and-privacy.html#pubin-path-pubcrate-pubsuper-and-pubself) syntax to restrict this to _just_ the system capsules?

Reply:
If a capsule can get a `ProcessBinary`, I'm not sure there's any harm in accessing it.

Reply:
I think we should really think of `pub(crate)` as serving two possible roles:

1. Protecting things that can break security (e.g. exposing a private field or allowing construction of a protected type) from other crates
2. Avoiding exposing things to other crates because it would make maintaining things harder if those methods/fields are not stable.

We don't really have a general story for (2) so I'm not super worried about this. I don't think (1) applies here.